### PR TITLE
Add Support For `ListBlockElement`

### PR DESCRIPTION
**Pull request by JamieB-gu**
Allows DCR to handle any CAPI `List` elements sent from frontend. It doesn't yet render anything for these elements, it just parses them into `KeyTakeaways` or `QAndAExplainer`.

The changes involved are:

- Adding `ListBlockElement` to the `FEElement` type and schema
- Adding `KeyTakeawaysBlockElement` to the `FEElement` type and schema
- Adding `QAndAExplainerBlockElement` to the `FEElement` type and schema
- Adding `enhanceLists` to convert from `ListBlockElement` to `KeyTakeawaysBlockElement` or `QAndAExplainerBlockElement`

**Note:** I recommend using "Hide whitespace" to view the diff.

### CODE REVIEW

File: dotcom-rendering/src/model/enhanceLists.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+								},
+							),
+						},
+					];
+				/**
+				 * If it's an unsupported list element, ignore the structure
+				 * and return the body elements.
+				 */
+				default:
+					return element.items.flatMap((i) => i.list);
```
Comment (davidfurey) [inline dotcom-rendering/src/model/enhanceLists.ts:37]: This seems reasonable to me

File: dotcom-rendering/src/model/enhanceLists.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	switch (element._type) {
+		case 'model.dotcomrendering.pageElements.ListBlockElement': {
+			switch (element.listElementType) {
+				case 'KeyTakeaways':
+					return [
+						{
+							_type: 'model.dotcomrendering.pageElements.KeyTakeawaysBlockElement',
+							keyTakeaways: element.items.flatMap(
+								({ title, list }) => {
+									if (title !== undefined) {
```
Comment (alinaboghiu) [inline dotcom-rendering/src/model/enhanceLists.ts:13]: perhaps a comment here to highlight if there is no title the element will be skipped/dropped/ignored?

File: dotcom-rendering/src/types/content.ts [hunk 3]
```diff
@@ -309,6 +309,27 @@ interface ItemLinkBlockElement {
...
+}
+
+interface KeyTakeawaysBlockElement {
+	_type: 'model.dotcomrendering.pageElements.KeyTakeawaysBlockElement';
+	keyTakeaways: KeyTakeaway[];
+}
+
+interface ListItem {
+	title?: string;
+	list: FEElement[];
```
[the remark this answers is not in the document]
Comment (davidfurey) [reply dotcom-rendering/src/types/content.ts:324]: It is called `elements` in the CAPI model, is that okay? I don't know where the name `list` came from

File: dotcom-rendering/src/model/enhanceLists.ts (hunk 2 above)
Comment (davidfurey) [reply dotcom-rendering/src/model/enhanceLists.ts:13]: Good idea. Once we have validation set up properly in composer for this, I suspect we will disallow publishing a key takeaway with a missing title

File: dotcom-rendering/src/types/content.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (alinaboghiu) [reply dotcom-rendering/src/types/content.ts:334]: elements is fine, sorry must have misread :D

**Review by alinaboghiu — APPROVED** [review]

**Pull request merged**